command/archive_tarball_create.go

```diff
@@ -31,9 +32,14 @@ type tarballCreate struct {
 	// e.g. "*.zip", "results.out", "ignore/**"
 	ExcludeFiles []string `mapstructure:"exclude_files" plugin:"expand"`
 
+	// This is only incremented in the case of a panic.
+	Attempt int
+
 	base
 }
 
+var retryErrors = []string{"out of range", "index > windowEnd"}
```

Comment:
IMO we should only do this for "index > windowEnd", since that only appears twice in the Go source code, both times in the function we think has a problem. "out of range" on the other hand could occur from nearly anywhere.

Reply:
Makes sense--we might still see other panics but we can log, hopefully fixing this will solve those as well.